### PR TITLE
chore(frameworks): Remove cucumber dependency on ConfigParser

### PR DESCRIPTION
**Pull request by sjelin**
First step towards splitting cucumber out (#2159)

### CODE REVIEW

File: lib/frameworks/cucumber.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
  * Execute the Runner's test cases through Cucumber.
  *
  * @param {Runner} runner The current Protractor Runner.
  * @param {Array} specs Array of Directory Path Strings.
+ * @param {function(string): string} resolvePath resolves a path relative to the
+ *     config file
  * @return {q.Promise} Promise resolved with the test results
  */
-exports.run = function(runner, specs) {
+exports.run = function(runner, specs, resolvePath) {
```
Comment (juliemr) [inline lib/frameworks/cucumber.js]: This seems like a messy way of passing this information.

The problem here is that some plugins or frameworks may want to add their own things to the configuration file which will need to be interpreted as file paths, which need to be resolved relative to something. This is fairly simple logic which the frameworks could probably do themselves, but we have an internal helper. Is it even worth exporting this internal helper? Most of the reason that `ConfigParser.resoveFilePatterns` is so long is ONLY to support cucumber stuff. 

I would say let's not export this function anywhere, simplify the internal implementation, and make cucumber do its own complicated logic with line numbers. What do you think?

Comment (sjelin) [inline lib/frameworks/cucumber.js]: Seams reasonable but we will have to export the config file's directory so cucumber can do its logic.

Comment (juliemr) [inline lib/frameworks/cucumber.js]: why? `config.configDir` isn't enough?

Comment (sjelin) [inline lib/frameworks/cucumber.js]: Ah, good point

**Comment by sjelin** [comment]
Canceling this commit, as it turns out we can remove cucumber without it.  Will simplify `ConfigParser` in the same commit as when we remove cucumber.

**Pull request closed without merging**